### PR TITLE
made buttons for graph

### PR DESCRIPTION
**Pull request by ingawei**
Screen Shot 2022-10-31 at 7 46 15 PM

### CODE REVIEW

**Comment by mqp** [comment]
1. Wouldn't it have been nicer to put these inside the box of the graph, maybe at the top right?
2. Why put these on the portfolio page but not on all graphs?

**Comment by ingawei** [comment]
> 1. Wouldn't it have been nicer to put these inside the box of the graph, maybe at the top right?
> 2. Why put these on the portfolio page but not on all graphs?

1. I think that'll lead to a lot of complications with the tooltip, also having it closer to the X axis makes sense since that's what it's modifying
2. I forget who but someone mentioned it might make sense for portfolio, and not markets. Markets tend to have a shorter lifetime/I imagine there's more reason for people to be interested in the long term patterns of a market, whereas most people want to see short term portfolio changes.

File: web/lib/icons/placeholder-graph.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+// from Feather: https://feathericons.com/
+export default function PlaceholderGraph(props: React.SVGProps<SVGSVGElement>) {
+  return (
+    <svg
+      xmlns="http://www.w3.org/2000/svg"
+      // width="18"
+      // height="18"
```
Comment (jahooma) [inline web/lib/icons/placeholder-graph.tsx:7]: Remove?

File: web/components/portfolio/portfolio-value-section.tsx [hunk 2]
```diff
@@ -1,111 +1,256 @@
...
     if (!portfolioHistory || !lastPortfolioMetrics) {
-      return <></>
+      return (
+        <PortfolioValueSkeleton
+          graphMode={graphMode}
+          onClickNumber={onClickNumber}
+          currentTimePeriod={currentTimePeriod}
+          setCurrentTimePeriod={(timePeriod: Period) =>
+            setCurrentTimePeriod(timePeriod)
+          }
```
Comment (jahooma) [inline web/components/portfolio/portfolio-value-section.tsx:40]: Could just be `setCurrentTimePeriod={setCurrentTimePeriod}`

File: web/components/portfolio/portfolio-value-section.tsx [hunk 3]
```diff
@@ -1,111 +1,256 @@
...
-          )}
-        </SizedContainer>
-      </>
+      <PortfolioValueSkeleton
+        graphMode={graphMode}
+        onClickNumber={onClickNumber}
+        currentTimePeriod={currentTimePeriod}
+        setCurrentTimePeriod={(timePeriod: Period) =>
+          setCurrentTimePeriod(timePeriod)
+        }
```
Comment (jahooma) [inline web/components/portfolio/portfolio-value-section.tsx:75]: Ditto

File: web/components/portfolio/portfolio-value-section.tsx [hunk 4]
```diff
@@ -1,111 +1,256 @@
...
+        </Row>
+      </Row>
+      <SizedContainer fullHeight={200} mobileHeight={100}>
+        {graphElement}
+      </SizedContainer>
+      <PortfolioTimeSelection
+        currentTimePeriod={currentTimePeriod}
+        setCurrentTimePeriod={(timePeriod: Period) =>
+          setCurrentTimePeriod(timePeriod)
+        }
```
Comment (jahooma) [inline web/components/portfolio/portfolio-value-section.tsx:186]: Ditto

**Review by jahooma — APPROVED** [review]
Cool, thanks Inga! LGTM.

Nice job with the skeletons. Code also looks good.

I do think the portfolio graph is the most important case to have the buttons and shorter default time range.

But it seems like the time range buttons would be valuable for the market page graphs too. Probably "All" is the best default for them, but I imagine many want to jump into the last 24 hours or 7 days to examine activity in detail, etc.

On mobile, there currently isn't a way to do that. And drag or pinch-to-zoom will never be that obvious.

I'm ok with that being a followup change though!

**Pull request merged**